app/controllers/concerns/project_filters.rb

```diff
@@ -44,8 +44,9 @@ def find_projects_by_params
   def find_forge_matches
     @match = Forge::Match.first(params[:codelocation])
     return unless @match
-    @projects = Project.where(id: Repository.matching(@match).joins(code_locations: :projects)
-                                    .select('projects.id')).not_deleted
+    matching_code_location_ids = CodeLocationSubscription.code_location_ids_matching_forge(@match)
+    matching_enlistments = Enlistment.where(code_location_id: matching_code_location_ids)
+    @projects = Project.where(id: matching_enlistments.pluck(:project_id)).not_deleted
```

Comment:
We can simplify this to single query
`Project.joins(:enlistments).where(enlistments: { code_location_id: matching_code_location_ids}).not_deleted`

Reply:
Good point